include/os/linux/spl/sys/cred.h

```diff
@@ -45,32 +45,73 @@ typedef struct cred cred_t;
 #define	SGID_TO_KGID(x)		(KGIDT_INIT(x))
 #define	KGIDP_TO_SGIDP(x)	(&(x)->val)
 
-static inline uid_t zfs_uid_into_mnt(struct user_namespace *mnt_ns, uid_t uid)
+extern boolean_t zfs_is_init_userns(struct user_namespace *ns);
+extern struct user_namespace *zfs_get_init_userns(void);
+
+static inline struct user_namespace *zfs_i_user_ns(struct inode *inode)
+{
+#ifdef HAVE_SUPER_USER_NS
+	return (inode->i_sb->s_user_ns);
+#else
+	return (NULL);
```

Comment:
Some old kernel (e.g. unpatched 3.10) does not have super_block->s_user_ns, but returning init_user_ns in that case makes sense. 

As the inline zfs_i_user_ns() will be used in zfs module (CDDL licensed) and the symbol "init_user_ns" is GPL, for this zfs_i_user_ns() to return &init_user_ns, we have to use a function zfs_get_init_userns() to get its pointer. My thought was to avoid function call if we can as it would be no fun to do tons of function calls when a busy server needs to handle lots of iops.

Reply:
I wonder if the perf impact is measurable. But just chatted with @sforshee and he pointed out that you could probably just stash the address in a local pointer at module init. The address won't change.

Reply:
Good idea. Will work on this.

Reply:
What we've done elsewhere is to grab the `init_user_ns` from `init_task.cred.user_ns` which we do have access to.  That's worked out reasonably well, I'd recommend that approach rather than introducing a new helpful function.  You can use `kcred->user_ns`.